lib/tasks/compare/ecf_users_and_induction_records.rake

```diff
@@ -6,67 +6,73 @@ namespace :compare do
-        puts "Duplicates found in induction record query:     #{duplicate_records.count}"
-        puts "Records found in both queries:                  #{same_records.count}"
-        puts "Records found users query but not IR:           #{users_but_not_ir.count}"
-        puts "Records found IR query but not users:           #{ir_but_not_users.count}"
-
         indexed_serialized_users = serialized_users[:data].index_by { |x| x[:id] }
         indexed_serialized_induction_records = serialized_induction_records[:data].index_by { |x| x[:id] }
 
-        puts "building detailed reports"
-
         users_with_no_ir_report = []
         ir_with_no_user_report = []
-        diff_report = ""
-
-        [*ids_from_serialized_users, *ids_from_serialized_induction_records].uniq.sort.each do |id|
-          record_from_users_query = indexed_serialized_users[id]
-          record_from_ir_query = indexed_serialized_induction_records[id]
+        diff_report = []
 
-          next if record_from_users_query == record_from_ir_query
+        ui_to_ei_map.each do |records|
+          record_from_users_query = indexed_serialized_users[records["id"]]
+          record_from_ir_query = indexed_serialized_induction_records[records["external_identifier"]]
 
-          if record_from_users_query.blank?
-            ir_with_no_user_report.push record_from_ir_query
+          if record_from_users_query.blank? && record_from_ir_query.blank?
+            # skip if both are not found - do we care about these ?
+            next
+          elsif record_from_users_query.blank?
+            ir_with_no_user_report.push user_id: records["id"], induction_record: record_from_ir_query
           elsif record_from_ir_query.blank?
-            users_with_no_ir_report.push record_from_users_query
+            users_with_no_ir_report.push external_identifier: records["external_identifier"], user_record: record_from_users_query
+          elsif record_from_users_query[:attributes] == record_from_ir_query[:attributes]
+            # skip if the attributes are the same
+            next
           else
-            diff_report += "####################\n"
-            diff_report += "ID: #{id}\n"
-            diff_report += JsonDiff.diff(record_from_users_query, record_from_ir_query, include_was: true).map { |el| "    #{el['path']}: \"#{el['was']}\"" }.join("\n")
-            diff_report += "--------------------\n"
-            diff_report += "#{JSON.pretty_generate(record_from_ir_query)}\n"
+            original_attributes = {}
+            JsonDiff.diff(record_from_users_query[:attributes], record_from_ir_query[:attributes], include_was: true).each do |el|
+              original_attributes[el["path"].sub("/", "")] = el["was"]
+            end
+
+            diff_report.push user_id: records["id"],
+                             external_identifier: records["external_identifier"],
+                             old_record: { attributes: original_attributes },
```

Comment:
is `old_record` the best description here? Aren't these the values that have been modified?